src/main/java/org/elasticsearch/index/translog/fs/FsTranslog.java

```diff
@@ -473,6 +474,33 @@ public long findLargestPresentTranslogId() throws IOException {
         }
     }
 
+    @Override
+    public OperationIterator openIterator(long translogId) throws IOException {
+        final Path translogName = getPath(translogId);
+        Path recoveringTranslogFile = null;
+        logger.trace("try open translog file {} locations: {}", translogName, Arrays.toString(locations()));
+        OUTER:
+        for (Path translogLocation : locations()) {
+            // we have to support .recovering since it's a leftover from previous version but might still be on the filesystem
+            // we used to rename the foo into foo.recovering since foo was reused / overwritten but we fixed that in 1.5
+            for (Path recoveryFiles : FileSystemUtils.files(translogLocation, translogName.getFileName() + "{.recovering,}")) {
+                logger.trace("translog file found in {}", recoveryFiles);
+                recoveringTranslogFile = recoveryFiles;
```

Comment:
I know this was copied over from another place, but I wonder if we should give preference to the recovering file. If I read this correctly , if we have both recovering and non-recovering, it is now random which one we choose.